lib/test-support/test-runner.js

```diff
@@ -25,47 +25,54 @@ module.exports = {
   'protocol': 'browser',
 };
 
-if (require.main === module) {
-  let path = require('path');
-  let url = require('url');
-  let fileUrl = require('file-url');
-  let treeKill = require('tree-kill');
-  let { start: efStart } = require('electron-forge');
+async function main() {
+  const path = require('path');
+  const url = require('url');
+  const fileUrl = require('file-url');
+  const treeKill = require('tree-kill');
+  const { api } = require('@electron-forge/core');
+  const { electronProjectPath } = require('ember-electron/lib/utils/build-paths');
 
-  let [, , buildDir, testemUrl, testPageUrl, id] = process.argv;
+  let [, , emberAppDir, testemUrl, testPageUrl, id] = process.argv;
 
   // The testPageUrl points to the testem server, so we need to turn it into a
   // file URL and add the testem ID to the query params.
-  let emberAppDir = path.join(buildDir, 'ember');
   let {
     pathname: testPagePath,
     query: testPageQuery,
   } = url.parse(testPageUrl, true);
   let indexPath = path.resolve(emberAppDir, path.join.apply(null, testPagePath.split('/')));
   let indexObj = url.parse(fileUrl(indexPath));
   indexObj.query = testPageQuery;
+
+  // https://github.com/nodejs/node/issues/9500
+  for (let [ key, value ] of Object.entries(indexObj.query)) {
+    if ([ null, undefined, '' ].includes(value)) {
+      indexObj.query[key] = 'true';
```

Comment:
Just to make sure I understand correctly, we want to work around the behavior described in the URL mentioned, so we interpret any query string parameter that doesn't have an associated value (`=...` / right-hand side) as meaning "this parameter is present" and thus assign it a value of "true" (sort of like [boolean HTML attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes#Boolean_Attributes)).

Reply:
Yeah, it's a kinda annoying bug in the URL library. When you assign a search object to a parsed URL, any keys whose values are the empty string (or undefined/null I think) end up doing this:

```javascript
const URL = require('url');

let obj = URL.parse('http://localhost/index.html', true);
obj.query = { hidepassed: '' };
console.log(URL.format(obj)); // http://localhost/index.html?hidepassed=
```

there's probably another workaround where we assign the query string as an un-parsed string, then round-trip format/parse it and then assign the testem id, but seems like more code/possibly less clear than this.